main.go

```diff
@@ -14,128 +10,235 @@ const GLOBAL_NAME = "sqlparser"
+				response["chainId"] = validTable.ChainID()
+				response["tableId"] = validTable.TokenID()
+				response["name"] = validTable.Name()
+			}
+			return resolve.Invoke(js.ValueOf(response))
+		}()
+		return nil
+	})
+	return Promise.New(handler)
+}
+
+func getUniqueTableNames(this js.Value, args []js.Value) interface{} {
+	Error := js.Global().Get("Error")
+	Promise := js.Global().Get("Promise")
+	if len(args) < 1 {
+		return Promise.Call("reject", Error.New("missing required argument: statement"))
+	}
+	statement := args[0].String()
+	handler := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		resolve := args[0]
+		reject := args[1]
+		go func() interface{} {
+			ast, err := sqlparser.Parse(statement)
+			if err != nil {
+				return reject.Invoke(Error.New("error parsing statement: " + err.Error()))
+			}
+			tableReferences := sqlparser.GetUniqueTableReferences(ast)
+			response := make([]interface{}, len(tableReferences))
+			for i := range tableReferences {
+				response[i] = tableReferences[i]
+			}
+			return resolve.Invoke(js.ValueOf(response))
+		}()
+		return nil
+	})
+	return Promise.New(handler)
+}
+
+func normalize(this js.Value, args []js.Value) interface{} {
+	Error := js.Global().Get("Error")
+	Promise := js.Global().Get("Promise")
+	if len(args) < 1 {
+		return Promise.Call("reject", Error.New("missing required argument: statement"))
+	}
+	statement := args[0].String()
+	handler := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		resolve := args[0]
+		reject := args[1]
+		go func() interface{} {
+			ast, err := sqlparser.Parse(statement)
+			// We always reject here on error, which means if the query has multiple statements
+			// with errors, only the first encountered error is thrown.
+			if err != nil {
+				return reject.Invoke(Error.New("error parsing statement: " + err.Error()))
+			}
+			if len(ast.Statements) == 0 {
+				return reject.Invoke(Error.New("error parsing statement: empty string"))
+			}
+			statements := make([]interface{}, len(ast.Statements))
+			var statementType StatementType
+			for i, stmt := range ast.Statements {
+				switch stmt.(type) {
+				case sqlparser.CreateTableStatement:
+					statementType = Create
+				case sqlparser.ReadStatement:
+					statementType = Read
+				case sqlparser.GrantOrRevokeStatement:
+					if statementType == "" {
+						statementType = Acl
 					}
-					return resolve.Invoke(js.ValueOf(node.StructureHash()))
-				}()
-				return nil
-			})
-			return js.Global().Get("Promise").New(handler)
-		}),
+				case sqlparser.WriteStatement:
+					statementType = Write
+				}
+				statements[i] = stmt.String()
+			}
+			response := map[string]interface{}{
+				"type":       string(statementType),
+				"statements": statements,
+			}
+			return resolve.Invoke(js.ValueOf(response))
+		}()
+		return nil
+	})
+	return Promise.New(handler)
+}
+
+func validateStatement(this js.Value, args []js.Value) interface{} {
```

Comment:
This is the workhorse function now. It basically checks everything _except_ interchain table mixing, which needs to be done on the client. If a statement passes this particular check, it is very likely going to work on the tableland network, at least from a syntax perspective.